### ISSUE TITLE
Vue Navigation Link Opening as Popup instead of New Page 🙈 Please Help

### ISSUE THREAD

**Issue by RudyBekker** [inertiajs/inertia-laravel#200]
Hey guys, I have a laravel application in Laravel Jetstream Vue..

When I click on a new navigation link I have created, this happens...

<img width="1433" alt="Screenshot 2020-12-26 at 11 48 22" src="https://user-images.githubusercontent.com/54132862/103150436-7afb4c80-4774-11eb-895b-c5a408dc222a.png">

it looks like its trying to open as a popup or something??? Maybe trying to treat it as a SPA link??? I don't know..

Here is my vue code.... I am following the same structure Taylor and them initially setup..

<img width="438" alt="Screenshot 2020-12-26 at 11 50 32" src="https://user-images.githubusercontent.com/54132862/103150447-8babc280-4774-11eb-9262-d5243a9a5d90.png">

Any help would be really appreciated

**Comment by RudyBekker** [comment]
I also checked the docs over at https://inertiajs.com/redirects#top

But I cannot find the correct method to work

**Comment by claudiodekker** [comment]
Hi @RudyBekker,

Since these [`jet-dropdown-link`](https://github.com/laravel/jetstream/blob/1.x/stubs/inertia/resources/js/Jetstream/DropdownLink.vue) are custom Jetstream components (meaning they're not part of Inertia itself) you won't find anything in our documentation on it. That said, since this specific component uses [`inertia-link`](https://inertiajs.com/links#creating-links) under the hood, I'm confident that what's going on here is either one of the following:
- You don't have a [named Laravel route](https://laravel.com/docs/8.x/urls#urls-for-named-routes) for `control`
- Ziggy (the thing that provides/handle the `route()` function you're using) isn't generaeting the route's URL properly.
- Some piece of back-end logic is causing your request to fail with a 404 ([`abort(404)`](https://laravel.com/docs/8.x/helpers#method-abort), [`Model::firstOrFail()`](https://laravel.com/docs/8.x/eloquent#not-found-exceptions), [non-existant route-model binding](https://laravel.com/docs/8.x/routing#implicit-binding) etc.)

My recommendation is to start by looking into your browser's `Console` or `Network` tabs to see if you can figure out whether your Inertia app is crashing for some reason (e.g. no required parameters passed to the route) and whether the request is correctly sent, because if both of those things _are_ correct, it means that your issue is originating from the Laravel (back-end) part of your app.

The reason I'm so confident of this, is because by design Inertia shows this pop-up (with the response's contents) when it isn't getting an Inertia response back. Usually I wouldn't be as sure of this as I am now, but because the content that you're seeing is a standard Laravel Nova `404` page (which is blade-rendered by default) this suddenly 'makes sense', because this is indeed not an Inertia response.

Hope this helps!

**Comment by RudyBekker** [comment]
@claudiodekker thanks for super detailed response! 

the vue component used is most certainly a inertia link 

<img width="843" alt="Screenshot 2020-12-26 at 13 25 02" src="https://user-images.githubusercontent.com/54132862/103151373-ebf33200-477d-11eb-97a9-2a23ce59da47.png">
 
Also worth mentironing that the Nova "popup" is a iframe...

My question is... how does one handle a redirect with inertia's routing methods?

<inertia-link href="/link">Link</inertia-link>

<inertia-link href="/link" method="**redirect??**">Logout</inertia-link>

--

the redirection guid on innteria has more emphasis on using a controller? 

I am just after a simple redirect?

they mention -> **return Inertia::location($url);**

but how do I use this in the .vue page or component?

kind regards
Rudy

**Comment by claudiodekker** [comment]
Hi Rudy,

It depends.. are you trying to **redirect to a page that isn't part of your website/Inertia app**? If so, you can just use an `<a>` tag instead of an `<inertia-link>`. Essentially, that's what `Inertia::location` is meant to do from the back-end.

As for making **redirects within your Inertia application**, that isn't really possible without hitting the back-end/controller, because in almost all situations the page that you're going to will need information to render it properly (are we allowed to view the page? Are we still logged in & as whom? Should we show a certain button? etc..)

(Theoretically, it is of course possible to hit some API instead, and to use it's response to do redirects etc. on the client-side, but this would mean that Inertia uses client-state instead of server-state, which just isn't how Inertia was designed to work.. Furthermore, you're not saving any calls really, because you'd still hit the API, just at a different moment)

In short, the rule is: When you're using an `inertia-link` to make a request, whatever happens internally MUST always return an `Inertia::render()`, [`Inertia::location()`](https://inertiajs.com/redirects#external-redirects) or 302 redirect to another route in your app (can be multiple even) that will eventually return one of these two.

Normally, with a logout, what you would do is hit the `logout` endpoint, process the actual logout, and then simply redirect the user back to the login page. This will then cause Inertia to load and render the login page, at which point your users see that.
